torch/utils/cpp_extension.py

```diff
@@ -1451,11 +1451,14 @@ def _get_build_directory(name, verbose):
             root_extensions_directory))
 
     build_directory = os.path.join(root_extensions_directory, name)
-    if not os.path.exists(build_directory):
-        if verbose:
-            print('Creating extension directory {}...'.format(build_directory))
+    try:
         # This is like mkdir -p, i.e. will also create parent directories.
         os.makedirs(build_directory, exist_ok=True)
+        if verbose:
+            print('Creating extension directory {}...'.format(build_directory))
+    except FileExistsError:
+        if verbose:
+            print('Using existing directory {}...'.format(build_directory))
```

Comment:
Hmm, this would never be executed, if exist_ok is set True, isn't it?
Also, `FileExistsError` would only be raised if python 3.7 or newer is used, while PyTorch still support Python-3.6, which would raise only OSError

Reply:
You're right. Then I think the race condition of making build dir does not exist anymore in the current master.